src/Parser/Mage/Fire/Modules/Items/MarqueeBindingsOfTheSunKing.js

```diff
@@ -0,0 +1,129 @@
+	}
+	
+	
+	// this section needs explaining:
+	// from looking through logs, the order of events is buff expires -> pyroblast cast begins within ~100ms -> pyroblast hits within 4.5 seconds
+	// i'm not sure this is the best way to accomplish this but it seems to work
+	
+	//get timestamp when buff expires
+	
+	on_byPlayer_removebuff(event) {
+    if (event.ability.guid !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+			return;
+		}
+		this.lastBuffTimestamp = this.owner.currentTimestamp;
+			}
+	
+	// for each pyroblast, check if buff expired within the past 100ms. if so, count it as buffed by the legendary
+	// additionally, set timestamp of cast to do a damage check
+	// i suppose it's possible for a player to instant cast pyroblast with hot streak within 100ms of the buff ending so it checks that this isn't the case.
+	
+	on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.PYROBLAST.id) {
+			return;
+		}
+    if (this.lastBuffTimestamp > this.owner.currentTimestamp - 100 && !this.combatants.selected.hasBuff(SPELLS.HOT_STREAK) ) {
+			this.buffedPyroblasts += 1;
+			this.lastPyroCastTimestamp = this.owner.currentTimestamp;
+		}
+		}
+
+	// after being cast, pyroblast takes a few seconds to travel and hit the target.
+	// allows 3 seconds for the pyroblast to hit, then resets timestamp so additional instant pyroblasts within the time limit don't count.
+	// this is messy and i don't know if there's an easier way
+	
+	on_byPlayer_damage(event) {
+    if (event.ability.guid !== SPELLS.PYROBLAST.id) {
+			return;
+		}
+		
+	if (this.owner.currentTimestamp - 3000 < this.lastPyroCastTimestamp)	
+			this.damage += event.amount + (event.absorbed || 0);
+			this.lastPyroCastTimestamp = 0;
+			console.log(this.damage);
+		}
+
+  item() {
+	  
+	  // i don't think this is the best section to put this code
+	  	this.expiredProcs = (this.legendaryProcs - this.overwrittenProcs - this.buffedPyroblasts);
+
+    return {
+		
+	 //	i debugged this and the console log is returning the expected damage done but the math for dps isn't working as i expect?
+	 // instead it returns something within about one % of the correct number
+		
+      item: ITEMS.MARQUEE_BINDINGS_OF_THE_SUN_KING,
+	  result: <ItemDamageDone amount={this.damage} />,
+    };
+  }
+  
+	// this is wordy, improve? also: change recommended numbers
+	// i couldn't find the icon for the bindings so i used pyroblast
+  
+    suggestions(when) {
+    when(this.overwrittenProcs).isGreaterThan(0)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span>Your buff from <ItemLink id={ITEMS.MARQUEE_BINDINGS_OF_THE_SUN_KING.id} /> was overwritten {formatPercentage(this.overwrittenProcs/this.legendaryProcs)}% of the time. This happens when the buff procs from an instant cast Pyroblast during Hot Streak before it is hard cast. If possible avoid using Fire Blast or Phoenix's Flames during Heating Up while you have the Marquee Bindings of the Sun King buff active.</span>)
```

Comment:
Please use `<Wrapper>` instead of `<span>`. Wrapper is basically the same thing but with much better render performance. I know you probably saw a lot of spans in the existing code, and we'll eventually clean it up, but for now the policy is "no new spans".

Get the import with
```
import Wrapper from 'common/Wrapper ';
```